commit 5ee14a4d543be4f502301cac136834475adc1899
Author: CODeRUS <coderusinbox@gmail.com>
Date:   Wed Feb 18 18:30:06 2015 +0500

    [v0.0.2] Added background with multiple recepient support

diff --git a/usr/share/jolla-messages/pages/ConversationPage.qml b/usr/share/jolla-messages/pages/ConversationPage.qml
index a2d4463..13a73ff 100644
--- a/usr/share/jolla-messages/pages/ConversationPage.qml
+++ b/usr/share/jolla-messages/pages/ConversationPage.qml
@@ -38,6 +38,33 @@ Page {
         onTargetChanged: textInput.reset()
     }
 
+    Row {
+    	z: -1
+        height: parent.height
+        spacing: 0
+
+        Repeater {
+        	model: conversation.people.length
+        	delegate: multipleAvatar
+        }
+    }
+
+    Component {
+    	id: multipleAvatar
+    	Item {
+    		width: conversationPage.width / conversation.people.length
+    		height: conversationPage.height
+    		clip: true
+		    Image {
+		        id: glassAvatar
+		        anchors.horizontalCenter: parent.horizontalCenter
+		        asynchronous: false
+		        sourceSize { width: conversationPage.width; height: conversationPage.isPortrait ? Screen.height : Screen.width }
+		        source: "image://glass/" + conversation.people[index].avatarPath
+		    }
+    	}
+    }
+
     MessagesView {
         id: messages
         focus: true
